evm-validate/src/lib.rs

```diff
@@ -1,8 +1,59 @@
+use std::collections::BTreeMap;
+
 use anyhow::{anyhow, Context, Result};
-use arrow::{
-    array::{Array, BinaryArray, UInt64Array},
-    record_batch::RecordBatch,
-};
+use arrow::array::{Array, BinaryArray, GenericByteArray, PrimitiveArray, UInt64Array, UInt8Array};
+use arrow::datatypes::{GenericBinaryType, UInt8Type};
+use arrow::{datatypes::UInt64Type, record_batch::RecordBatch};
+
+use alloy_primitives::{Address, Bloom, Bytes, FixedBytes, Log, PrimitiveSignature, TxKind, Uint, B256, U256};
+use alloy_consensus::{Eip658Value, Receipt, ReceiptEnvelope, TxEip1559, TxEip2930, TxEip4844, TxEip4844Variant, TxEnvelope, TxLegacy, SignableTransaction};
+use alloy_consensus::proofs::{calculate_transaction_root, calculate_receipt_root};
+use alloy_eips::eip2930::{AccessList, AccessListItem};
+use serde::Deserialize;
+
+struct LogArray<'a> {
```

Comment:
don't think these are needed. Can use them as dynamically typed RecordBatch and get columns one by one like so:

let block_number = record_batch.get_column("block_number").as_any().downcast_ref().unwrap();

Reply:
Not sure I understand your complain. These are exactly what I'm doing in the extract functions (bottom of the file). I'm using these structs to not use long tuples in function return.